### PR TITLE
FIX: Raise errors when wrong string arguments are passed to `resample`

### LINKED ISSUE #19303: resample() silently ignores misstyped str kwargs, e.g. for label
#### Code Sample, a copy-pastable example if possible

```python
In [1]: import pandas as pd

In [2]: df = pd.DataFrame(index=pd.date_range(start='2010-01-23', periods=1000, freq='min'), data={'foo': pd.np.random.rand(1000)})

In [3]: df.resample('h').mean().tail()
Out[3]: 
                          foo
2010-01-23 12:00:00  0.477997
2010-01-23 13:00:00  0.488294
2010-01-23 14:00:00  0.426095
2010-01-23 15:00:00  0.549081
2010-01-23 16:00:00  0.466148

In [4]: df.resample('h', label='right').mean().tail()
Out[4]: 
                          foo
2010-01-23 13:00:00  0.477997
2010-01-23 14:00:00  0.488294
2010-01-23 15:00:00  0.426095
2010-01-23 16:00:00  0.549081
2010-01-23 17:00:00  0.466148

In [5]: 

In [5]: df.resample('h', label='right_wrong').mean().tail()
Out[5]: 
                          foo
2010-01-23 12:00:00  0.477997
2010-01-23 13:00:00  0.488294
2010-01-23 14:00:00  0.426095
2010-01-23 15:00:00  0.549081
2010-01-23 16:00:00  0.466148

# The last command should fail!

```
#### Problem description

Recently (and maybe many times before without noticing...) I mistyped the string passed as kwarg `label` in `resample`, e.g. `df.resample('5min', label='rigth')` and wondered later why the alignment of my resampled data with the raw data was wrong. It turns out that the actual string passed as kwarg `label` is never checked to be either `left` or `right`. Hence, if it is 

Is this on purpose, or should this be fixed?

If yes, a check could be inserted [here](https://github.com/pandas-dev/pandas/blob/ca2d2611c5a657720661990e2f3016356e577d1d/pandas/core/resample.py#L1065). Or maybe deeper in the class hierarchy? Maybe other kwargs, like `closed` could also be affected. 

If I get a pointer I could come up with a PR.

#### Output of ``pd.show_versions()``

<details>

```
INSTALLED VERSIONS
------------------
commit: None
python: 2.7.14.final.0
python-bits: 64
OS: Darwin
OS-release: 16.7.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: de_DE.UTF-8
LOCALE: None.None

pandas: 0.21.0
pytest: 3.2.5
pip: 9.0.1
setuptools: 36.7.2
Cython: None
numpy: 1.13.3
scipy: 1.0.0
pyarrow: 0.7.1
xarray: 0.10.0
IPython: 5.5.0
sphinx: None
patsy: 0.4.1
dateutil: 2.6.1
pytz: 2017.3
blosc: None
bottleneck: 1.2.1
tables: None
numexpr: None
feather: 0.4.0
matplotlib: 2.1.0
openpyxl: None
xlrd: 1.1.0
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: 0.999999999
sqlalchemy: None
pymysql: None
psycopg2: 2.7.3.2 (dt dec pq3 ext lo64)
jinja2: 2.9.6
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
```

</details>

### PR DESCRIPTION
**Pull request by cchwala**
- [x] closes #19303
- [x] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

As a first step, the initial commit just extends the resampling tests so that they fail when a misstyped string argument, e.g. for `label`, passes silently without raising.

I have the fix for the code ready and will commit when the tests have failed once.

### CODE REVIEW

File: pandas/tests/test_resample.py [hunk 1]
```diff
@@ -963,6 +963,15 @@ def test_resample_basic(self):
         rng = date_range('1/1/2000 00:00:00', '1/1/2000 00:13:00', freq='min',
                          name='index')
         s = Series(np.random.randn(14), index=rng)
+
```
Comment (jreback) [inline pandas/tests/test_resample.py]: can you add as a separate test.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/test_resample.py (hunk 1 above)
Comment (cchwala) [reply pandas/tests/test_resample.py]: Yes. No problem.

**Comment by jreback** [comment]
tests look good. pls push the fix when you can.

File: pandas/core/resample.py [hunk 2]
```diff
@@ -1061,6 +1061,16 @@ class TimeGrouper(Grouper):
     def __init__(self, freq='Min', closed=None, label=None, how='mean',
                  axis=0, fill_method=None, limit=None, loffset=None,
                  kind=None, convention=None, base=0, **kwargs):
+        # Check for correctness of the keyword arguments which would
+        # otherwise silently use the default if misspelled
+        if label not in {None, 'left', 'right'}:
+            raise AttributeError('Unsupported value %s for `label`' % label)
```
Comment (TomAugspurger) [inline pandas/core/resample.py]: These should probably be `ValueError`s.

File: doc/source/whatsnew/v0.23.0.txt [hunk 3]
```diff
@@ -485,7 +485,7 @@ Groupby/Resample/Rolling
 
 - Bug when grouping by a single column and aggregating with a class like ``list`` or ``tuple`` (:issue:`18079`)
 - Fixed regression in :func:`DataFrame.groupby` which would not emit an error when called with a tuple key not in the index (:issue:`18798`)
--
+- Bug in :func:`DataFrame.resample` which silently ignored unsupported (or misstyped) options for `label`, `closed` and `convention` (:issue:`19303`)
```
Comment (TomAugspurger) [inline doc/source/whatsnew/v0.23.0.txt]: Use double backticks for `label`, `closed` and `convention`.

**Review by TomAugspurger — APPROVED** [review]
Looks good @cchwala, ping us when the tests pass.

File: pandas/core/resample.py [hunk 4]
```diff
@@ -1061,6 +1061,16 @@ class TimeGrouper(Grouper):
...
                  axis=0, fill_method=None, limit=None, loffset=None,
                  kind=None, convention=None, base=0, **kwargs):
+        # Check for correctness of the keyword arguments which would
+        # otherwise silently use the default if misspelled
+        if label not in {None, 'left', 'right'}:
+            raise ValueError('Unsupported value %s for `label`' % label)
+        if closed not in {None, 'left', 'right'}:
+            raise ValueError('Unsupported value %s for `closed`' % closed)
+        if convention not in {None, 'start', 'end', 'e', 's'}:
+            raise ValueError('Unsupported value %s for `convention`'
```
Comment (jreback) [inline pandas/core/resample.py]: don't use '%', use ``.format()`` instead.

File: pandas/tests/test_resample.py [hunk 5]
```diff
@@ -985,6 +986,19 @@ def test_resample_basic(self):
         expect = s.groupby(grouper).agg(lambda x: x[-1])
         assert_series_equal(result, expect)
 
+    def test_resample_string_kwargs(self):
+        rng = date_range('1/1/2000 00:00:00', '1/1/2000 00:13:00', freq='min',
```
Comment (jreback) [inline pandas/tests/test_resample.py]: add the issue number here as a comment

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/test_resample.py [hunk 6]
```diff
@@ -985,6 +986,19 @@ def test_resample_basic(self):
...
+                         name='index')
+        s = Series(np.random.randn(14), index=rng)
+
+        # Check that wrong keyword argument strings raise an error
+        with pytest.raises(ValueError) as e_info:
```
Comment (cchwala) [inline pandas/tests/test_resample.py]: @jreback There is also a linting error here because of the unused `e_info`. I will also remove these, okay?

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @cchwala

**Pull request merged**